realm/src/main/java/io/realm/DynamicRealm.java

```diff
@@ -0,0 +1,283 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package io.realm;
+
+import android.os.Looper;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.CopyOnWriteArrayList;
+
+import io.realm.exceptions.RealmException;
+import io.realm.internal.Table;
+import io.realm.internal.TableView;
+
+/**
+ * DynamicRealm is a dynamic variant of {@link io.realm.Realm}. This means that all access to data and/or queries are
+ * done using string based class names instead of class type references.
+ *
+ * The same {@link io.realm.RealmConfiguration} can be used to open a Realm file in both dynamic and typed mode, but
+ * modifying the schema while having both a typed and dynamic version open is highly discouraged and will most likely
+ * crash the typed Realm. During migrations only a DynamicRealm will be open.
+ *
+ * Dynamic Realms do not enforce schemas and schema versions and {@link RealmMigration} code is not used even if it
+ * has been defined in the {@link RealmConfiguration}.
+ *
+ * @see io.realm.Realm
+ */
+public class DynamicRealm extends BaseRealm {
+
+    private static final ThreadLocal<Map<RealmConfiguration, DynamicRealm>> realmsCache =
+            new ThreadLocal<Map<RealmConfiguration, DynamicRealm>>() {
+                @Override
+                protected Map<RealmConfiguration, DynamicRealm> initialValue() {
+                    return new HashMap<RealmConfiguration, DynamicRealm>();
+                }
+            };
+
+    private static final ThreadLocal<Map<RealmConfiguration, Integer>> referenceCount =
+            new ThreadLocal<Map<RealmConfiguration,Integer>>() {
+                @Override
+                protected Map<RealmConfiguration, Integer> initialValue() {
+                    return new HashMap<RealmConfiguration, Integer>();
+                }
+            };
+
+    private DynamicRealm(RealmConfiguration configuration, boolean autoRefresh) {
+        super(configuration, autoRefresh);
+    }
+
+    /**
+     * Realm static constructor that returns a dynamic variant of the Realm instance defined by provided
+     * {@link io.realm.RealmConfiguration}. Dynamic Realms do not care about schemaVersion and schemas, so opening a
+     * DynamicRealm will never trigger a migration
+     *
+     * @return The DynamicRealm defined by the configuration.
+     * @see RealmConfiguration for details on how to configure a Realm.
+     */
+    public static DynamicRealm getInstance(RealmConfiguration configuration) {
+        if (configuration == null) {
+            throw new IllegalArgumentException("A non-null RealmConfiguration must be provided");
+        }
+        return create(configuration);
+    }
+
+    /**
+     * Instantiates and adds a new object to the Realm.
+     *
+     * @param className The class of the object to create.
+     * @return The new object.
```

Comment:
And other capitalized first characters in this file. see example https://github.com/realm/realm-wiki/wiki/Java-Doc-guidelines